### DISCUSSION TITLE
can we add a custom script inside vvvebjs ? after initialization ? programmatically ?

### DISCUSSION THREAD

**Discussion by syedketto** [givanz/VvvebJs#300] [Q&A] [answered]
Hi, 
great project,
i was wondering whether we can add a custom script which can create html elements like select option on typing a $ the select gets opened and we can select from option
for that to work 
can we add a custom script inside vvvebjs ? after initialization ? programmatically ?
If yes please give me some steps
I'm integrating it inside react so I don't need to add script using <script src='myscript'> etc.
rather I am looking to import script inside the file where vvveb.js is initialized and that from that initialized instance i want to add script or any alternative can be a better option.
Please Help!

**Comment by givanz** [comment, marked as answer]
Hi

Thank you.
You want to create new components or add custom js scripts in the page after is already loaded?

You can run javascript code after the editor loads the page from here
https://github.com/givanz/VvvebJs/blob/master/editor.html#L1708

Or you can run code after the iframe is loaded with `Vvveb.Builder.documentFrame.on("load", function(e)  { });`

To dynamically add components or sections after the editor has finished loading you can call the usual  `Vvveb.Components.add()`  https://github.com/givanz/VvvebJs/wiki/Components 

Then you will need to call `Vvveb.Builder.loadControlGroups() ` to reload the components after adding your components.
For sections and blocks you need to call `Vvveb.Builder.loadSectionGroups()` or `Vvveb.Builder.loadBlockGroups()` respectively.

If you want to create new select input you can check the current select input implementation from  
https://github.com/givanz/VvvebJs/blob/master/libs/builder/inputs.js#L116-L132
https://github.com/givanz/VvvebJs/blob/master/editor.html#L920-L946

https://github.com/givanz/VvvebJs/wiki/Inputs

**Reply by syedketto** [reply]
I was needing in imports and export kind of thing so that we can use it inside React 
Nevertheless,
I will figure out the rest,
Thanks Brother,
Great work again
